tests/tablenavigation.js

```diff
@@ -0,0 +1,2284 @@
+
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[ '11', '12' ],
+					[ '[]', '' ]
+				] ) );
+			} );
+
+			it( 'should not create another row and not move the caret if the "insertTableRowBelow" command is disabled', () => {
+				setModelData( model, modelTable( [
+					[ '11', '12[]' ]
+				] ) );
+
+				const insertTableRowBelowCommand = editor.commands.get( 'insertTableRowBelow' );
+
+				insertTableRowBelowCommand.forceDisabled( 'test' );
+
+				editor.editing.view.document.fire( 'keydown', domEvtDataStub );
+
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[ '11', '12[]' ]
+				] ) );
+			} );
+
+			it( 'should move to the first cell of the next row if at the end of a row', () => {
+				setModelData( model, modelTable( [
+					[ '11', '12[]' ],
+					[ '21', '22' ]
+				] ) );
+
+				editor.editing.view.document.fire( 'keydown', domEvtDataStub );
+
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[ '11', '12' ],
+					[ '[21]', '22' ]
+				] ) );
+			} );
+
+			it( 'should move to the next table cell if the block content is partially selected', () => {
+				setModelData( model, modelTable( [
+					[ '11', '<paragraph>12</paragraph><paragraph>[foo]</paragraph><paragraph>bar</paragraph>', '13' ]
+				] ) );
+
+				editor.editing.view.document.fire( 'keydown', domEvtDataStub );
+
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[
+						'11',
+						'<paragraph>12</paragraph><paragraph>foo</paragraph><paragraph>bar</paragraph>',
+						'[13]'
+					]
+				] ) );
+			} );
+
+			it( 'should move to the next cell containing an image', () => {
+				setModelData( model, modelTable( [
+					[ '11[]', '<paragraph>foo</paragraph><image><caption></caption></image>' ]
+				] ) );
+
+				editor.editing.view.document.fire( 'keydown', domEvtDataStub );
+
+				sinon.assert.calledOnce( domEvtDataStub.preventDefault );
+				sinon.assert.calledOnce( domEvtDataStub.stopPropagation );
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[ '11', '<paragraph>[foo</paragraph><image><caption></caption></image>]' ]
+				] ) );
+			} );
+
+			it( 'should move to the next cell containing a block quote', () => {
+				model.schema.register( 'blockQuote', {
+					allowWhere: '$block',
+					allowContentOf: '$root'
+				} );
+				editor.conversion.elementToElement( { model: 'blockQuote', view: 'blockquote' } );
+
+				setModelData( model, modelTable( [
+					[ '11[]', '<blockQuote><paragraph>foo</paragraph></blockQuote>' ]
+				] ) );
+
+				editor.editing.view.document.fire( 'keydown', domEvtDataStub );
+
+				sinon.assert.calledOnce( domEvtDataStub.preventDefault );
+				sinon.assert.calledOnce( domEvtDataStub.stopPropagation );
+				assertEqualMarkup( getModelData( model ), modelTable( [
+					[ '11', '<blockQuote><paragraph>[foo]</paragraph></blockQuote>' ]
+				] ) );
+			} );
+
+			it( 'should listen with the lower priority than its children', () => {
```

Comment:
I don't understand the description of this test. By reading the code I understand this is related to event bubbling and canceling but it's not clear what are the children and whose children they are. Maybe there's a way to re-phrase it?

Reply:
This is only moved from `TableEditing` tests. I wasn't even reading those.

Reply:
I guess it's indentation handler.

Reply:
That's on me probably and it is a moved code. Let's leave it for now.